src/webapp/src/auth/asyncAuthActions.tsx

```diff
@@ -0,0 +1,29 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { AuthError } from '@firebase/auth-types';
+import { auth, provider } from '../firebase';
+import { UserData } from './types';
+
+export const signInUser = createAsyncThunk<
+  UserData,
+  void,
+  { rejectValue: AuthError }
+>('user/sigInUser', async (_, { rejectWithValue }) => {
+  try {
+    const { user } = await auth.signInWithPopup(provider)!;
+    const { displayName, uid, email } = user!;
```

Comment:
Can we do that?
```suggestion
    const { user: { displayName, uid, email } } = await auth.signInWithPopup(provider)!;
```

Reply:
hi @paschalidi, 
I wanted to do it like that but I couldn't! Without asserting that `user` is not null by doing `user!` the compiler complains.